### ISSUE TITLE
[machine learning] save state of model between train and test commands

### ISSUE THREAD

**Issue by codejohnson** [KiwiCampusChallenge/Kiwi-Campus-Challenge#20]
One of the requirements of the solution is the ability to run separated commends for train and test:
python app.py train -m [chosen model] -d [directory with trainig data]
python app.py test -m [chosen model] -d [directory with test data]

Due this is my first approach to a solution of machine learning (I never touch this paradigm), I´m having an intense self-learning experience working on the solution. 
But I´m having problems trying to imagine if I need to save a model memory state between the two commands, because the train command will terminate the execution of the console application, and a subsequent execution of test session will not have a trained model available. 
Should I save the state of the learning model after finishing the train command, and, after that, being able to have a test session?
I have now a first solution using LR, but using an unsolicited command that train and test the model without splitting the session:

python app.py trainandtest -m [chosen model] -d [directory with test data]

Im having around 91%-94% accuracy

thank you.

**Comment by KiwiCampusChallenge** [comment]
Great question, nice to see that people are learning from this challenge!

you can save your model as a [pickle](http://scikit-learn.org/stable/modules/model_persistence.html) file, as a TensorFlow graph or a numpy arrays. The most important thing is that you save it when you run your `train` command and restore the model when you use your `test` command, as well as the infer command.
